searchlib/src/tests/attribute/stringattribute/stringattribute_test.cpp

```diff
@@ -434,21 +383,29 @@ StringAttributeTest::testSingleValue(Attribute & svsa, Config &cfg)
     load.load();
 }
 
-
-
-int
-StringAttributeTest::Main()
+TEST("testSingleValue")
 {
-    TEST_INIT("stringattribute_test");
-
-    testMultiValue();
-    testMultiValueMultipleClearDocBetweenCommit();
-    testMultiValueRemove();
-    testSingleValue();
+    EXPECT_EQUAL(24u, sizeof(AttributeVector::SearchContext));
+    EXPECT_EQUAL(96u, sizeof(SingleValueStringAttribute::StringSingleImplSearchContext));
+    {
+        Config cfg(BasicType::STRING, CollectionType::SINGLE);
+        SingleValueStringAttribute svsa("svsa", cfg);
+        const IAttributeVector * ia = &svsa;
+        EXPECT_TRUE(dynamic_cast<const SingleValueEnumAttributeBase *>(ia) != nullptr);
+        testSingleValue(svsa, cfg);
 
-    TEST_DONE();
-}
+        SingleValueStringAttribute svsb("svsa", cfg);
+        testDefaultValueOnAddDoc(svsb);
+    }
+    {
+        Config cfg(BasicType::STRING, CollectionType::SINGLE);
+        cfg.setFastSearch(true);
+        SingleValueStringPostingAttribute svsa("svspb", cfg);
+        testSingleValue(svsa, cfg);
 
+        SingleValueStringPostingAttribute svsb("svspb", cfg);
+        testDefaultValueOnAddDoc(svsb);
+    }
 }
 
-TEST_APPHOOK(search::StringAttributeTest);
+TEST_MAIN() { TEST_RUN_ALL(); }
```

Comment:
Consider adding a newline at end of file.

Reply:
Fixed